packages/medusa/src/services/event-bus.ts

```diff
@@ -165,106 +194,60 @@ export default class EventBusService {
   async emit<T>(
     eventName: string,
     data: T,
-    options: EmitOptions = {}
-  ): Promise<StagedJob | void> {
-    if (this.transactionManager_) {
-      const stagedJobRepository = this.transactionManager_.getCustomRepository(
-        this.stagedJobRepository_
-      )
-
-      const stagedJobInstance = stagedJobRepository.create({
-        event_name: eventName,
-        data,
-      } as StagedJob)
-      return await stagedJobRepository.save(stagedJobInstance)
-    } else {
-      const opts: { removeOnComplete: boolean } & EmitOptions = {
-        removeOnComplete: true,
-      }
-      if (typeof options.attempts === "number") {
-        opts.attempts = options.attempts
-        if (typeof options.backoff !== "undefined") {
-          opts.backoff = options.backoff
-        }
-      }
-      if (typeof options.delay === "number") {
-        opts.delay = options.delay
-      }
-      this.queue_.add({ eventName, data }, opts)
+    options: EmitOptions & { cacheKey?: string } = {}
+  ): Promise<void> {
+    // construct options for the job
+    const opts: { removeOnComplete: boolean } & EmitOptions = {
+      removeOnComplete: true,
     }
-  }
-
-  startEnqueuer(): void {
-    this.shouldEnqueuerRun = true
-    this.enqueue_ = this.enqueuer_()
-  }
-
-  async stopEnqueuer(): Promise<void> {
-    this.shouldEnqueuerRun = false
-    await this.enqueue_
-  }
-
-  async enqueuer_(): Promise<void> {
-    while (this.shouldEnqueuerRun) {
-      const listConfig = {
-        relations: [],
-        skip: 0,
-        take: 1000,
+    if (typeof options.attempts === "number") {
```

Comment:
todo: NaN is a number, but still it is not valid. I would suggest for those type of checks to create a new utils `isValidNumber` which could check that it is different from NaN and of type number